model-server-openapi/specifications/model-server.yaml

```diff
@@ -13,481 +13,700 @@ info:
-            type: string
-        - name: Sec-WebSocket-Key
-          in: "header"
-          required: true
-          description: "Websocket Sec-WebSocket-Key parameter"
-          schema:
-            type: string
-      responses:
-        "403":
-          $ref: '#/components/responses/403'
-        "401":
-          $ref: '#/components/responses/401'
-        "500":
-          $ref: '#/components/responses/500'
-        "101":
-          description: "Switching Protocols"
-          headers:
-            Connection:
-              required: true
-              schema:
-                type: string
-            Upgrade:
-              required: true
-              schema:
-                type: string
-            Sec-WebSocket-Accept:
-              required: true
+          $ref: "#/components/responses/VersionDelta"
+        "404":
+          description: Repository, branch or `from` hash does not exist.
+          content:
+            application/problem+json:
               schema:
-                type: string
-  /v2/repositories/{repository}/branches/{branch}/poll:
-    get:
-      operationId: pollRepositoryBranch
-      parameters:
-        - name: lastKnown
-          in: "query"
-          required: false
-          schema:
-            type: string
-        - name: repository
-          in: "path"
-          required: true
-          schema:
-            type: string
-        - name: branch
-          in: "path"
-          required: true
-          schema:
-            type: string
+                $ref: "#/components/schemas/Problem"
+        default:
+          $ref: "#/components/responses/Problem"
+
+  # TODO specify input and output formats
+  /repositories/{repository}/branches/{branch}/query:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+      - $ref: "#/components/parameters/BranchPath"
+
+    post:
+      operationId: postRepositoryBranchQuery
+      tags:
+        - ModelQl access
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
       responses:
-        "403":
-          $ref: '#/components/responses/403'
-        "401":
-          $ref: '#/components/responses/401'
-        "500":
-          $ref: '#/components/responses/500'
         "200":
-          $ref: '#/components/responses/versionDelta'
-  /v2/repositories/{repository}/branches/{branch}/pollHash:
+          $ref: '#/components/responses/200json'
+        default:
+          $ref: '#/components/responses/Problem'
+
+  /repositories/{repository}/versions/{versionHash}/delta:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+      - $ref: "#/components/parameters/VersionHashPath"
     get:
-      operationId: pollRepositoryBranchHash
+      operationId: getRepositoryVersionHash
+      description: |-
+        If `from` is not specified, this **returns the full version data**.
+      tags:
+        - versions
```

Comment:
`/repositories/{repository}/versions/{versionHash}/delta` does not need `versions` tag. `delta access` is enough. I think the `versions` tag is not needed to cluster endpionts that operate on a version, because those endpoints operate very differntly on the versions. (e.g. delta-access, model-access, Model-QL access)

Reply:
That tag was intended to collect everything that operates on a single version. Doesn't that make sense?

Reply:
It is odd because `version` is the only tag that tags endpoints from different mechanisms.